### DISCUSSION TITLE
Keybase meeting Thursday, April 1st 21h00 UTC

### DISCUSSION THREAD

**Discussion by huey735** [bisq-network/bisq#5378] [General]
[Previous meeting discussion](https://github.com/bisq-network/bisq/discussions/5322)

Topics for the next meeting:
	
A. JAVA UX developper
+ What exactly would this developer be workinging on?
+ What's stopping Bisq from being a mobile or web app? A clarification on the specif vision for it is necessary. What would this mobile/web app do?
+ [Early mockups for a new Bisq UI](https://github.com/bisq-network/bisq-design/blob/master/bisq-refresh.pdf) by @pedromvpg
+ [Why does Bisq need a new GUI?](https://github.com/bisq-network/bisq/discussions/5371)

B. New release - https://github.com/bisq-network/bisq/releases/tag/v1.6.2

+ Highlights
			
	- API
		+ Proposal - https://github.com/bisq-network/proposals/issues/136#issuecomment-554443705 - This is an old project that aims to bring more automated functionality to Bisq
		+ Project issue - https://github.com/bisq-network/projects/issues/11
		+ Project board - https://github.com/orgs/bisq-network/projects/17#card-56865611
		
	- Initial working on Segwit BSQ - https://github.com/bisq-network/bisq/pull/5000

# 

This is a new format for me so I'd like to establish some broad rules
- The person typing has the floor. If you see someone typing, do not type. You can start typing elsewhere (text editor).
- Be as concise and compact as possible. Make an effort of writing all you have to write at the time in a single block of text and posting it at once. Do not break a thought in multiple messages.
- React to what's being said with emojis to express agreement (:+1:), disagreement (:-1: ), confusion (:question:) and any other that efficiently describes your sentiment.
- If you have question consider giving different emojis as the options so that the attendants can vote with the emojis to voice their opinion without having to write them down.

Please feel free to make any suggestions bellow.

**Comment by pazza83** [comment]
Hi @huey735 please can a discussion take place about how to address the burningman problem in one of the team meetings. 

There are a number of options for example: https://github.com/bisq-network/proposals/issues/316

**Comment by pazza83** [comment]
I have made the proposal comment for voting for the new UI to get started here: https://github.com/bisq-network/projects/issues/49#issuecomment-812277419  Please give a quick thumbs up or thumbs down. On the Github comment. Thanks

**Comment by m52go** [comment]
## Meeting Minutes

Agenda:
```
Today we will be following up on some topics discussed on the last 
meeting (https://github.com/bisq-network/bisq/discussions/5322) plus 
touching up on some highlights of the new release:

	A. New UI (https://github.com/bisq-network/projects/issues/49)
		+ What exactly would a new UI developer be working on?
		+ What's stopping Bisq from being a mobile or web app? A 
                  clarification on the specific vision for it is necessary. What
                  would this mobile/web app do?
		+ Why does Bisq need a new GUI? 
                  (https://github.com/bisq-network/bisq/discussions/5371)
		+ Early mockups for a new Bisq UI by @pedromvpg
                  (https://github.com/bisq-network/bisq-design/blob/master/bisq-refresh.pdf)
		
	B. New release (https://github.com/bisq-network/bisq/releases/tag/v1.6.2)
                  - API
                          + Proposal (https://github.com/bisq-network/proposals/issues/136) - this 
                            is a project that aims to bring more automated functionality to Bisq
                          + Issue (https://github.com/bisq-network/projects/issues/11)
                          + Project board (https://github.com/orgs/bisq-network/projects/17)
				
                  - Initial working on Segwit BSQ (https://github.com/bisq-network/bisq/pull/5000)
```

### New UI (https://github.com/bisq-network/projects/issues/49)

There was some initial disagreement over whether or not investing in a new UI for the existing Bisq software is a good move.

Some said there are other objectives which should have higher priority than a UI overhaul (e.g. reducing on-chain footprint), and many agreed, but the UI overhaul doesn't overlap much with those other objectives while making Bisq a whole lot more approachable for new users. Overall it seemed that there was good consensus for moving forward with the UI overhaul, which was confirmed with up-votes [on the project proposal](https://github.com/bisq-network/projects/issues/49).

Protocol development could use more hands, but finding such developers is not easy...so might as well allow the UI work to continue in parallel.

Costs for UI overall will be better known once the UI developer takes a closer look at the code through performance testing and designing the onboarding UI (these are examples of initial tasks to see how well things are working out).

Full project proposal is linked above, but some key ideas were listed in the chat:

- Assess if JavaFX is necessary
- Refactor the code to streamline future changes to the UI
- Add an onboarding flow (completely new)
- Work on improving the UI to make the client more up-to-date and elegant
    - Standardize styles and clean up inconsistencies
    - Move nav to the side
    - Rework responsiveness

Timeline:
https://docs.google.com/spreadsheets/d/1RvjngdWhKX8L47tLkEP9eiJjfSb-SyUWvHnDYZP5YcQ/edit?usp=sharing

Mockup:
https://xd.adobe.com/view/a83c2327-4730-4ec2-8938-e318b2749588-fd6f

### What's stopping Bisq from being a mobile or web app?

API needs to serve JSON to enable web and mobile use cases, which is something API developer would prefer to explore later when API is more mature and more used...but could happen sooner if it turns out gRPC-web is feasible to use.

Control of current notification apps is in the process of being handed over to a more active Bisq developer. This new developer is planning to fix a couple of high-priority bugs, but probably won't be doing any significant development on the apps.

### Segwit BSQ (https://github.com/bisq-network/bisq/pull/5000)

Initial code shipped in 1.6.0 for activation toward the end of May and full delivery in the next planned release (if all goes well).

Developer leading efforts on atomic BSQ-BTC swaps is hoping to have an update on his efforts for meeting in 2 weeks...this effort involves major refactorings, but will set the stage for a single-tx trade protocol.

---

Potential topic for next meeting: Bisq DAO privacy and improving the colored coin implementation.
- Video summary (https://arxiv.org/abs/2007.07048)
- Paper (https://www.youtube.com/watch?v=VRr-aWQpAWA)

**Reply by Conza88** [reply]
Great notes, thanks!

**Comment by pazza83** [comment]
Hi @huey735 and @m52go 

Not sure if it is just me but I am finding it a little hard to follow along with the team meeting; when they are next due, where the agenda is published and where the outcome are. For example I am unsure if I have missed a meeting since 1st April?

I think the above is due to the following:

- team-meeting channel in Keybase has no pinned information as to a team meeting repository, and conversations are deleted after 24 hours.
- GitHub Discussions while great for some things have some limitations. Mainly things ordered by votes as opposed to chronically? an linked issues referenced not showing up on said issues. 

Therefore I would like to propose the following:

- Team-Meeting repository to be created https://github.com/bisq-network/team-meeting
- Team meeting repository referenced as a pinned message in Keybase team meeting.
- Each team meeting to be given it's own issue in the team meeting repository
- Maybe creation of an upcoming team meeting issue to list dates / times of upcoming issues and allow for the suggestion of new topics for discussion?

I think the team meeting have been great. Hopefully the above or something similar will make it easier for things reference to be included in the relevant issues, and make it easier for users to quickly find any information needed.

Let me know your thoughts / suggestions.

**Reply by huey735** [reply]
Currently the meetings are announced here on Discussions. The meeting takes place over Keybase. The notes of the meeting are shared here after by @m52go .

But yes I agree with most of your suggestions. Plus I like that issues can be closed which doesn't seem possible with Discussions topics.

Maybe we can use an existing repository to test your suggestions? We have:
* :zero: -  https://github.com/bisq-network/projects
* :one: : -  https://github.com/bisq-network/events
* :two: -  https://github.com/bisq-network/forum
